OutOfSchool/OutOfSchool.BusinessLogic/Services/ContactsService.cs

```diff
@@ -0,0 +1,113 @@
+using System.Diagnostics.CodeAnalysis;
+using AutoMapper;
+using OutOfSchool.BusinessLogic.Models;
+using OutOfSchool.BusinessLogic.Models.ContactInfo;
+using OutOfSchool.Services.Models.ContactInfo;
+
+namespace OutOfSchool.BusinessLogic.Services;
+
+public class ContactsService<TEntity, TDto> : IContactsService<TEntity, TDto>
+    where TEntity : BusinessEntity, IHasContacts
+    where TDto : IHasContactsDto<TEntity>
+{
+    private readonly IMapper mapper;
+
+    public ContactsService(IMapper mapper)
+    {
+        this.mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+    }
+
+    public void ProcessCreate(TEntity existingEntity, TDto dto)
```

Comment:
`ProcessCreate/Update` name is too vague and promising. Consider narrowing it with declaring intentions more clearly, i.e. `PrepareNew/UpdatedContacts`.